### PR TITLE
[3.x] Fix TileSet editor selected regression

### PR DESCRIPTION
**Pull request by Mickeon**
Fixes https://github.com/godotengine/godot/issues/88618 (Actually tested)

### CODE REVIEW

**Review by lawnjelly — APPROVED** [review]

File: editor/plugins/tile_set_editor_plugin.cpp [hunk 1]
```diff
@@ -3573,30 +3573,30 @@ void TilesetEditorContext::_get_property_list(List<PropertyInfo> *p_list) const
...
+		p_list->push_back(PropertyInfo(Variant::OBJECT, PNAME("tile_material"), PROPERTY_HINT_RESOURCE_TYPE, "ShaderMaterial"));
+		p_list->push_back(PropertyInfo(Variant::COLOR, PNAME("tile_modulate")));
+		p_list->push_back(PropertyInfo(Variant::INT, PNAME("tile_tile_mode"), PROPERTY_HINT_ENUM, "SINGLE_TILE,AUTO_TILE,ATLAS_TILE"));
+		if (tileset->tile_get_tile_mode(id) == TileSet::AUTO_TILE) {
+			p_list->push_back(PropertyInfo(Variant::INT, PNAME("tile_autotile_bitmask_mode"), PROPERTY_HINT_ENUM, "2x2,3x3 (minimal),3x3"));
+			p_list->push_back(PropertyInfo(Variant::VECTOR2, PNAME("tile_subtile_size")));
+			p_list->push_back(PropertyInfo(Variant::INT, PNAME("tile_subtile_spacing"), PROPERTY_HINT_RANGE, "0, 1024, 1"));
+		} else if (tileset->tile_get_tile_mode(id) == TileSet::ATLAS_TILE) {
+			p_list->push_back(PropertyInfo(Variant::VECTOR2, PNAME("tile_subtile_size")));
+			p_list->push_back(PropertyInfo(Variant::INT, PNAME("tile_subtile_spacing"), PROPERTY_HINT_RANGE, "0, 1024, 1"));
```
Comment (kleonc) [inline editor/plugins/tile_set_editor_plugin.cpp:3593]: Please re-add this line you've removed in #64931 when rebasing from https://github.com/godotengine/godot/commit/29a69cb51e260fb7be3a4d0bd2396c06ea0beb40 to https://github.com/godotengine/godot/commit/211c707e67725f27ae995a923dd206cca9976315. Originally it was added in #71533.
And please be more careful when rebasing / force pushing. :upside_down_face:

```suggestion
			p_list->push_back(PropertyInfo(Variant::INT, PNAME("tile_subtile_spacing"), PROPERTY_HINT_RANGE, "0, 1024, 1"));
			p_list->push_back(PropertyInfo(Variant::INT, PNAME("tile_autotile_fallback_mode"), PROPERTY_HINT_ENUM, "Auto,Icon"));
```

**Review by kleonc — CHANGES_REQUESTED** [review]

File: editor/plugins/tile_set_editor_plugin.cpp (hunk 1 above)
Comment (Mickeon) [reply editor/plugins/tile_set_editor_plugin.cpp:3593]: It had been a years-old PR, ow 💦💦💦

Comment (kleonc) [reply editor/plugins/tile_set_editor_plugin.cpp:3593]: Sorry but I've suggested it on a wrong line. :laughing: It should be within `if (tileset->tile_get_tile_mode(id) == TileSet::AUTO_TILE)` block.

Comment (Mickeon) [reply editor/plugins/tile_set_editor_plugin.cpp:3593]: We're all collectively f\*\*\*\*ups

**Comment by Mickeon** [comment]
Back where it should be. This time I checked https://github.com/godotengine/godot/pull/64931 to be sure.

**Review by kleonc — APPROVED** [review]
Seems strange to me that the block adding `Selected Tile` section is after the block handling `TileSetEditor::EDITMODE_COLLISION` but before the blocks handling `TileSetEditor::EDITMODE_NAVIGATION` / `TileSetEditor::EDITMODE_OCCLUSION`.
But seems like `selected_*` property is shown within the inspector at the begining regardless of this so I guess it's fine as is... :thinking:

||||
|--------|--------|--------|
|godot windows opt tools 64_L9H9zjqcM8|O18VhvDk4S|17ADzunuRM|

**Comment by Mickeon** [comment]
The original intention was to keep the `Collision Options` group (`selected_collision`) close to the `selected_collision` property, as the property shares the same prefix and it cannot be included in the group.

**Comment by lawnjelly** [comment]
Thanks!

**Pull request merged**